tests/unit/api/test_security.py

```
@@ -0,0 +1,151 @@
+    os.environ.pop("AUTH__ENABLED", None)
+    get_settings.cache_clear()
+
+    assert verify_bearer_token(None) is None
+    assert verify_bearer_token("Bearer anything") is None
+
+
+@pytest.mark.unit
+def test_auth_enabled_accepts_valid_token(reset_auth_env: None) -> None:
+    """A matching bearer token passes when auth is enabled."""
+    os.environ["AUTH__ENABLED"] = "true"
+    os.environ["AUTH__BEARER_TOKEN"] = "s3cret"
```

Ruff (0.15.14)

[error] Possible hardcoded password assigned to: "AUTH__BEARER_TOKEN"

(S105)